staging/src/k8s.io/controller-manager/app/options/types.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	componentbaseconfig "k8s.io/component-base/config"
+)
+
+// ServiceControllerConfiguration contains elements describing ServiceController.
```

Comment:
Can we put the controller configs in sub-directories. Separating controller concerns from controller-manager concerns seems prudent. It seems fine to have service controller here, while its jsut service controller. However when it becomes, service, cloud, cloud lifecycle, route, x, y and z controllers, I think we will want them each in their own sub directory.